plugins/bigquery/dbt/adapters/bigquery/connections.py

```diff
@@ -267,6 +267,50 @@ def execute(self, sql, auto_begin=False, fetch=None):
 
         return status, res
 
+    # The following method intentionaly violates DRY, in that it is mostly
+    # copy-pasted from raw_execute().  This is done in order to discorage
+    # use of legacySQL queries in DBT, except to obtain partition metadata.
+    # the method would be removed when partition metadata becomes available
+    # from standardSQL.
+    def _raw_execute_legacy_sql(self, sql, fetch=False):
+        conn = self.get_thread_connection()
+        client = conn.handle
+
+        logger.debug('On {}: {}', conn.name, sql)
+
+        job_params = {'use_legacy_sql': True}
+
+        priority = conn.credentials.priority
+        if priority == Priority.Batch:
+            job_params['priority'] = google.cloud.bigquery.QueryPriority.BATCH
+        else:
+            job_params[
+                'priority'] = google.cloud.bigquery.QueryPriority.INTERACTIVE
+
+        maximum_bytes_billed = conn.credentials.maximum_bytes_billed
+        if maximum_bytes_billed is not None and maximum_bytes_billed != 0:
+            job_params['maximum_bytes_billed'] = maximum_bytes_billed
+
+        def fn():
+            return self._query_and_results(client, sql, conn, job_params)
+
+        query_job, iterator = self._retry_and_handle(msg=sql, conn=conn, fn=fn)
+
+        return query_job, iterator
+
+    def get_partitions_metadata(self, table_id):
```

Comment:
IMO this method should take a [relation](https://docs.getdbt.com/reference/dbt-classes/#relation) as its argument, instead of a string (`table_id`). This change would mean that:
* We can construct the legacy SQL table reference using relation components, rather than relying on `from_string()`
* Users can pass a `ref()`, `source()`, or relation object to the Jinja macro directly. I expect the most common use case (incremental models) to call this as `get_partitions_metadata(this)`.